src/react-native/11-navigation/navigation.md

```diff
@@ -64,11 +161,73 @@ If you’ve implemented the solution correctly, the tests will pass when you run
 
 ## Objective 2: Using React Native stack navigation
 
-TODO
-
-### Concept TODO
-
-TODO
+Now that we've covered tab navigation, let's move on to stack navigation. As mentioned in the previous section, most applications use a combination of both tab and stack navigation to create a seamless user experience.
+
+### Stack vs Native Stack
+
+The React Navigation library provides both a stack navigator as well as a native stack navigator. The native stack navigator uses the native APIs provided by the platform, `UINavigationController` on iOS and `Fragment` on Android. Because it is using the native APIs, it provides native performance and exposes native-specific features. Although the stack navigator is not as performant, it is an easier-to-use, more customizable, JavaScript-based alternative.
+
+### Performance
+
+Both the stack navigator and tab navigator have an optional prop, `detachInactiveScreens`, that defaults to `true`. This boolean value indicates whether inactive screens (screens that are not currently visible) should be detached from the view hierarchy. This can improve performance by reducing the number of views that need to be rendered and maintained in memory.
+
+### Stack Navigator
+
+Let's take a look at an example of using a stack navigator in React Native with React Navigation.
+
+First, let's setup our application to use stack navigation from React Navigation. To do this, we will install the `@react-navigation/stack` package:
+
+```bash
+npm install @react-navigation/stack
+```
+
+```tsx
+import { createStackNavigator } from '@react-navigation/stack';
+
+const Stack = createStackNavigator();
+
+function MyStack() {
+  return (
+    <Stack.Navigator
+      initialRouteName="Home"
+      screenOptions={{
+        headerMode: 'screen',
+        headerTintColor: 'white',
+        headerStyle: { backgroundColor: 'tomato' },
+      }}
+    >
+      <Stack.Screen
+        name="Home"
+        component={Home}
+        options={{
+          title: 'Awesome app',
+        }}
+      />
+      <Stack.Screen
+        name="Profile"
+        component={Profile}
+        options={{
+          title: 'My profile',
+        }}
+      />
+      <Stack.Screen
+        name="Settings"
+        component={Settings}
+        options={{
+          gestureEnabled: false,
+        }}
+      />
+    </Stack.Navigator>
+  );
+}
+```
+@highlight 1, 3, 7-36
+
+Let's break down the code above:
+
+**createStackNavigator**
+
+Similar to the `createBottomTabNavigator` function, the `createStackNavigator` function creates a navigator that renders a stack of screens. The components returned from this function are the `Stack.Navigator` and `Stack.Screen` components which we use together to define our stack navigation. Unlike the tab navigator, we do not need to wrap our stack navigator in a `NavigationContainer` component.
```

Comment:
I think we need to explain a little more here. Taking a quick look at the exercise and solution for this objective, they need to know how to:

- Understand `header` in `screenOptions`, and what `navigation.canGoBack()` does.
- Know what the `component` and `name` props to `.Screen` do.

Reply:
also fleshed this one out a lot more